File: org.kevoree.modeling.microframework/src/test/java/org/kevoree/modeling/microframework/test/HelperTest.java
package org.kevoree.modeling.microframework.test;

import org.junit.Test;
import org.junit.Assert;
import org.kevoree.modeling.api.util.Helper;


/**
 * Created by duke on 10/13/14.
 */
public class HelperTest {

    @Test
    public void helperTest() {
        Assert.assertNull(Helper.parentPath("/"));
        Assert.assertNull(Helper.parentPath(null));
        Assert.assertNull(Helper.parentPath(""));
        Assert.assertEquals("/", Helper.parentPath("/nodes[name=n0]"));
        Assert.assertEquals("/nodes[name=n0]",Helper.parentPath("/nodes[name=n0]/children[name=c4]"));
    }

}

File: org.kevoree.modeling.microframework.typescript/src/test/ts/org/kevoree/modeling/microframework/test/HelperTest.ts
///<reference path="../../../../junit/Test.ts"/>
///<reference path="../../api/util/Helper.ts"/>
// TODO Resolve static imports
///<reference path="../../../../junit/Assert.ts"/>
// TODO Resolve static imports
///<reference path="../../../../junit/Assert.ts"/>
// TODO Resolve static imports
///<reference path="../../../../junit/Assert.ts"/>

class HelperTest {

  public helperTest(): void {
    assertNull(Helper.parentPath("/"));
    assertNull(Helper.parentPath(null));
    assertNull(Helper.parentPath(""));
    assertEquals("/", Helper.parentPath("/nodes[name=n0]"));
    assertEquals("/nodes[name=n0]", Helper.parentPath("/nodes[name=n0]/children[name=c4]"));
  }

}

